### ISSUE TITLE
Engine crashes when loading region files on game starts on macOS(m1)

### ISSUE THREAD

**Issue by Richard74Huang** [Zylann/godot_voxel#481]
VoxelEngine crashes when loading region files when VoxelEngine::thread_count > 1.

godot_voxel : 676c101f
godot 4 : af48b296e6
system: macOS 13.0.1
Apple silicon

 Following are crash info:

[1] 1   libsystem_platform.dylib            0x000000018c7282a4 _sigtramp + 56
[2] String::copy_from(char const*) (in godot.macos.editor.arm64) + 44
[3] String::String(char const*) (in godot.macos.editor.arm64) + 24
[4] Variant::Variant(char const*) (in godot.macos.editor.arm64) + 36
[5] zylann::voxel::gd::to_dict(zylann::voxel::VoxelEngine::Stats::ThreadPoolStats const&) (in godot.macos.editor.arm64) + 336
[6] zylann::voxel::gd::to_dict(zylann::voxel::VoxelEngine::Stats const&) (in godot.macos.editor.arm64) + 64
[7] zylann::voxel::gd::VoxelEngine::get_stats() const (in godot.macos.editor.arm64) + 44
[8] Image::set_as_black()
[9] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (in godot.macos.editor.arm64) + 122212
[10] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (in godot.macos.editor.arm64) + 348
[11] Node::_notification(int) (in godot.macos.editor.arm64) + 1808

There are more than 5000 load block tasks were running and it was very difficult to pin down the crash point.

By setting thread_count to 1, it works well.

**Comment by Zylann** [comment]
The crash info isn't a faulty area, so the problem is likely somewhere else. Also not sure why `Image::set_as_black` gets in there, it makes no sense.
I don't know how to proceed from here because the problem could really be anywhere.

**Comment by Richard74Huang** [comment]
Sorry I didn't mention about my testing project, I'm testing the engine with your blocky_game demo. I think there is a clue of what causing the crash. If I comment out VoxelEngine.get_state() in debug_info.gd, thread_count > 1 will work just fine. 

I guess that there is a thread from GDScript that also trying to access status of VoxelEngine simultaneously.... just a guess.

**Comment by Zylann** [comment]
The demo does not spawn any thread from GDScript. Even if there was, `get_stats` is a reading operation, so there should not be problems calling it from multiple threads.

**Comment by Richard74Huang** [comment]
No luck with VoxelStreamSQLite :(  It also crashes if get_stats() get called. But by removing the stream, it always works well with get_states().

I noticed that it always crashes while accessing **thread_debug_task_name**   even if those threads were still busy picking tasks up and their debug_task_name were blank or pointed to a strange address. So I modified get_thread_debug_task_name:

`const char *ThreadedTaskRunner::get_thread_debug_task_name(unsigned int thread_index) const {
	if(_threads[thread_index].debug_state == STATE_RUNNING) {
		return _threads[thread_index].debug_running_task_name;
	}
	else {
		return nullptr;
	}
}`

This avoid the crash(on m1 Mac) .

**Comment by MGilleronFJ** [comment]
Instead of doing that, what happens if you change this line in `threaded_task_runner.h`:
```cpp
std::atomic<const char *> debug_running_task_name = nullptr;
```

**Comment by Zylann** [comment]
Should be fixed in ac634fd670d376f1da0e41b2af4d2849c74a5f50

Thank you for investigating :)

**Comment by Zylann** [comment]
Closing as fixed
